### ISSUE TITLE
CopyButton does not announce state change with screen readers

### ISSUE THREAD

**Issue by emyarod** [carbon-design-system/carbon#5385]
## Environment

macOS, Chrome 79, Voiceover

## Detailed description

related https://github.com/carbon-design-system/carbon/issues/2523

> the CopyButton Component is also affected by this issue. "copied" is not announced after the button is pressed.

from https://github.com/carbon-design-system/carbon/issues/2523#issuecomment-566209585

## Steps to reproduce the issue

Navigate to the `<CopyButton>` story and interact with the component with a screen reader. The change in copy button state is not announced when the button is clicked.
